### ISSUE TITLE
Shiny App grey's out with socket error

### ISSUE THREAD

**Issue by sascha42** [openanalytics/shinyproxy#233]
Hi guys 

I’m running a dockerized ShinyProxy on an linux machine.

The docker image runs perfectly fine if we run it in our company where we have a very good and fast (1Gbit)
ethernet connection.
But if somebody run it where the ethernet is not that fast (home office via vpn) suddenly the screen turns to gray
if the user is updating a large plot (40000 datapoints).
In the browser (pushing F12) we get the following error message.

![image](https://user-images.githubusercontent.com/71326048/93217130-e2aefe00-f768-11ea-9e52-1ef9500de7f7.png)

In the "shiny-proxy" debugg log we get the following error message:
2020-09-15 12:11:01.778 DEBUG 1 --- [   XNIO-2 I/O-8] i.u.client.http.HttpClientConnection     : connection to 67a9fca528ee/172.19.0.3:3838 closed.

So after long testing we found out that this behavior is showing up if the shiny-proxy is sending the data for the plot to the browser the same time the browser is sending the heardbeat to the shiny-proxy.

This behavior is very easy to reproduce if we set the "heartbeat-rate" to very short period like 2 seconds. We also set the "heartbeat-timeout" to very long period like 10 minutes with the same results.

To solve the problem for now we sett's the heartbeat-rate to an extremly high value (1 year) but it's not realy acceptable for long terms.
Is there any possibility to fix this issue?

Here is the code that we used to reproduce the problem in a very efficient way especially if you push the button for updating the plot very often in quick succession (that's just for testing, for production we use real data ;-)).

Thanks in advance.

#--------------------------------------------------------------------------
library(shiny)
library(plotly)
library(shinycssloaders)

#--------------------------------------------------------------------------
ui <- fluidPage(
  column(
    width = 9,
    plotlyOutput("scatterplot") %>% withSpinner()
  ),
  
  column(
    width = 3,
    numericInput("nr_points", label = "Count random numbers", min = 1, value = 300000),
    actionButton("update", label = "Update Scatterplot")
  )
)

#--------------------------------------------------------------------------
server <- function(input, output, session) {

  output$scatterplot <- renderPlotly({
    input$update

    isolate({
      xdat <- runif(n = input$nr_points)
      ydat <- runif(n = input$nr_points)
      plot_ly(x = xdat, y = ydat, type = "scattergl", mode = "markers")
    })
  })
}

app <- shinyApp(ui, server)
runApp(app, host = '0.0.0.0', port = 3838)

**Comment by LEDfan** [comment]
Hi @sascha42 

Thank you for your report and research! We can reproduce the issue and are looking into a solution for the next release.

**Comment by sascha42** [comment]
Hi @LEDfan
 
Thank you for your fast response.
I tried the docker container „ledfan/oa-shinyproxy-snapshot:2.4.1-feature-23249 https://hub.docker.com/layers/ledfan/oa-shinyproxy-snapshot/2.4.1-feature-23249/images/sha256-e7b48f7388440f49097a1f3188606e7c6d7236974876d0eff07bd723d833856d?context=explore" that you recommended in our environment, but when „Siny-Proxy“ try to start a „R“ container we get
 
the following error with the error code 500.
 
eu.openanalytics.containerproxy.ContainerProxyException: Failed to start container
 Caused by: java.io.IOException: Permission denied
at jnr.unixsocket.UnixSocketChannel.doConnect(UnixSocketChannel.java:127)
 
We never had that problem with our standard „Shiny Proxy“ image.
Please give me some advice.
 
Thanks
 
 
 
 
 
 
 
Von: Tobia De Koninck <notifications@github.com> 
Gesendet: Dienstag, 13. Oktober 2020 16:31
An: openanalytics/shinyproxy <shinyproxy@noreply.github.com>
Cc: sascha42 <sascha.steiner@predictive.at>; Mention <mention@noreply.github.com>
Betreff: Re: [openanalytics/shinyproxy] Shiny App grey's out with socket error (#233)
 
Hi @sascha42 https://github.com/sascha42
I implemented a possible fix for this issue in the following commit: openanalytics/containerproxy@9668a13 https://github.com/openanalytics/containerproxy/commit/9668a137a680eedff1e9aadb431c18777dd316d5
In my tests the websocket connection did not break anymore even with very low heartbeat rates (500ms) and spamming the Update Scatterplot .
Could you please give this a test in your environment?
Here is a docker container with the fix included: ledfan/oa-shinyproxy-snapshot:2.4.1-feature-23249 https://hub.docker.com/layers/ledfan/oa-shinyproxy-snapshot/2.4.1-feature-23249/images/sha256-e7b48f7388440f49097a1f3188606e7c6d7236974876d0eff07bd723d833856d?context=explore
Thanks

**Comment by LEDfan** [comment]
Hi @ sascha42 

It's strange that you are facing this issue. I just tried to run this specific container and I cannot reproduce the error.
In order to debug the issue, can you explain a bit your setup. E.g. give the command you use to run the command, your `application.yml` and `/etc/docker/daemon.json` files.

Thanks

**Comment by sascha42** [comment]
Hi LEDfan
 
Here ist he dockerfile that is usually use, not in this case because i used your image:
#-------------------------------------------------------------------------------------------------------------------------------------
FROM openjdk:8-jre
 
RUN mkdir -p /opt/shinyproxy/
RUN wget https://www.shinyproxy.io/downloads/shinyproxy-2.3.0.jar -O /opt/shinyproxy/shinyproxy.jar
 
COPY application.yml /opt/shinyproxy/application.yml
WORKDIR /opt/shinyproxy/
CMD ["java", "-jar", "/opt/shinyproxy/shinyproxy.jar"]
#-------------------------------------------------------------------------------------------------------------------------------------
 
Here is the run command that is used:
#------------------------------------------------------------------------------------------------------------------------------------------
docker run --name Shinyproxy_ledfan -d \
-v /var/run/docker.sock:/var/run/docker.sock \
-v /root/Visualytics_Working_Copy_2020_08_24/ShinyProxy_Image/application.yml:/opt/shinyproxy/application.yml \
--net sp-example-net \
-p 8090:8080 c3856833d393
#----------------------------------------------------------------------------------------------------------------------------------------
The „c3856833d393" is the id of the image.
 
Here is the „application.yml“ 
#--------------------------------------------------------------------------------
server:
  useForwardHeaders: true
  servlet.session.timeout: 0
proxy:
  port: 8080
  hide-navbar: true
  heartbeat-rate: 20000
  heartbeat-timeout: 20000
  container-wait-time: 30000
  container-log-path: ./container-logs
  authentication: simple
  admin-groups: admin
  users:
  - name: testuser1
    password: testuser1
    groups: admin
  - name: testuser2
    password: testuser2
    groups: worker
 
 
  docker:
      internal-networking: true
  specs:
  - id: visualytics
    display-name: Visualytics
    container-cmd: ["R","root/visualytics/"]
    container-image: rstarter
    container-network: sp-example-net
    container-volumes: [ "/root/Visualytics_Working_Copy_2020_08_24/ShinyProxy_Image/rFiles:/root/visualytics" ]
    container-env:
      DATA_MONGO_URL: mongodb://10.0.0.41:27017
      USER_MGMT_MONGO_URL: mongodb://10.0.0.41:27017/UserMGMT
    access-groups: [admin,worker]
 
 
  logging:
    file:
      shinyproxy.log
    level:
      root: DEBUG
 
#---------------------------------------------------------------------------------------------
And here is the error message:
 
Error
Status code: 500
Message: Failed to start container
Stack Trace:
eu.openanalytics.containerproxy.ContainerProxyException: Failed to start container
at eu.openanalytics.containerproxy.backend.AbstractContainerBackend.startProxy(AbstractContainerBackend.java:113)
at eu.openanalytics.containerproxy.service.ProxyService.startProxy(ProxyService.java:213)
at eu.openanalytics.shinyproxy.controllers.AppController.getOrStart(AppController.java:107)
at eu.openanalytics.shinyproxy.controllers.AppController.startApp(AppController.java:64)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878)
at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792)
at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:517)
at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
at io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)
at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:126)
at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:90)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:118)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:158)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.authentication.www.BasicAuthenticationFilter.doFilterInternal(BasicAuthenticationFilter.java:155)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.csrf.CsrfFilter.doFilterInternal(CsrfFilter.java:117)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:92)
at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:77)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)
at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)
at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:93)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
at io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
at io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
at io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
at io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
at io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
at io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
at io.undertow.server.handlers.PathHandler.handleRequest(PathHandler.java:91)
at eu.openanalytics.containerproxy.util.ProxyMappingManager$ProxyPathHandler.handleRequest(ProxyMappingManager.java:160)
at io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
at io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:132)
at io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
at io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
at io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
at io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
at io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
at io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:269)
at io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:78)
at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:133)
at io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:130)
at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
at io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:249)
at io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:78)
at io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:99)
at io.undertow.server.Connectors.executeRootHandler(Connectors.java:370)
at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2019)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1558)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1449)
at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.spotify.docker.client.exceptions.DockerException: java.util.concurrent.ExecutionException: javax.ws.rs.ProcessingException: java.io.IOException: Permission denied
at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2848)
at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2723)
at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:742)
at com.spotify.docker.client.DefaultDockerClient.createContainer(DefaultDockerClient.java:724)
at eu.openanalytics.containerproxy.backend.docker.DockerEngineBackend.startContainer(DockerEngineBackend.java:84)
at eu.openanalytics.containerproxy.backend.AbstractContainerBackend.doStartProxy(AbstractContainerBackend.java:129)
at eu.openanalytics.containerproxy.backend.AbstractContainerBackend.startProxy(AbstractContainerBackend.java:110)
... 106 more
Caused by: java.util.concurrent.ExecutionException: javax.ws.rs.ProcessingException: java.io.IOException: Permission denied
at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2721)
... 111 more
Caused by: javax.ws.rs.ProcessingException: java.io.IOException: Permission denied
at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:528)
at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:535)
at org.glassfish.jersey.client.ClientRuntime.lambda$null$6(ClientRuntime.java:181)
at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:288)
at org.glassfish.jersey.client.ClientRuntime.lambda$createRunnableForAsyncProcessing$7(ClientRuntime.java:155)
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
... 1 more
Caused by: java.io.IOException: Permission denied
at jnr.unixsocket.UnixSocketChannel.doConnect(UnixSocketChannel.java:127)
at jnr.unixsocket.UnixSocketChannel.connect(UnixSocketChannel.java:136)
at jnr.unixsocket.UnixSocketChannel.connect(UnixSocketChannel.java:223)
at com.spotify.docker.client.UnixConnectionSocketFactory.connectSocket(UnixConnectionSocketFactory.java:85)
at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)
at org.glassfish.jersey.apache.connector.ApacheConnector.apply(ApacheConnector.java:480)
... 14 more

**Comment by LEDfan** [comment]
Hi @sascha42

Thank you for the information, I now understand the problem. Our official Docker containers are running ShinyProxy using a non-root account, therefore it has no access to the docker socket.

I updated the image on docker hub to run SP as root inside the container: ledfan/oa-shinyproxy-snapshot:2.4.1-feature-23249-root (image name has changed!).
The id should now be 3816d4dcd82a.

I'm looking forward to you test results!

**Comment by sascha42** [comment]
Hi @LEDfan
 
After two days of heavy testing in our environment I can report the following:
 
With 1000ms, 2000ms and 5000ms the application grays out again after updating the plot several times. However,  with 20000ms, which I think is a reasonable setting, there was no more grey out.
 
So it looks like you fixed it.
 
Congratulations and thank you for your great support.
 
Can you tell me when the fix will be implemented in an official rollout of „Shinyproxy“.
 
Thanks a lot

**Comment by LEDfan** [comment]
Hi @sascha42 

I'm happy to hear that the issue is (almost) solved.
I adapted the code a bit so that a least a delay of 5 seconds is used. This should make the heartbeat more reliable with smaller intervals. I created a new docker image for this `ledfan/oa-shinyproxy-snapshot:2.4.1-feature-23249-root-extra-fix`.
Feel free to test this.

Regarding the official release, this fix will of course be included in the next version, but we don't know yet when we will effectively release it.
